x-pack/plugins/security_solution/public/management/components/endpoint_execute_action/execute_action_host_response_output.tsx

```diff
@@ -91,50 +87,78 @@ const ExecutionActionOutputAccordion = memo<ExecuteActionOutputProps>(
 );
 ExecutionActionOutputAccordion.displayName = 'ExecutionActionOutputAccordion';
 
-interface ExecuteActionHostResponseOutputProps {
+export interface ExecuteActionHostResponseOutputProps {
   action: MaybeImmutable<ActionDetails>;
   agentId?: string;
   'data-test-subj'?: string;
   textSize?: 's' | 'xs';
 }
 
 export const ExecuteActionHostResponseOutput = memo<ExecuteActionHostResponseOutputProps>(
-  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize }) => {
+  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize = 'xs' }) => {
     const outputContent = useMemo(
       () =>
         action.outputs &&
         action.outputs[agentId] &&
         (action.outputs[agentId].content as ResponseActionExecuteOutputContent),
-      [agentId, action]
+      [action.outputs, agentId]
     );
 
+    const { data: actionDetails, isFetching } = useGetActionDetails(action.id, {
+      enabled: !outputContent,
+    });
+
+    const [executeOutputContent, setExecuteOutputContent] = useState<
+      undefined | ResponseActionExecuteOutputContent
+    >(outputContent);
+
+    useEffect(() => {
+      if (
+        !isFetching &&
+        actionDetails &&
+        actionDetails.data &&
+        actionDetails.data.outputs &&
+        actionDetails.data.outputs[agentId]
+      ) {
+        setExecuteOutputContent(
+          actionDetails.data.outputs[agentId].content as ResponseActionExecuteOutputContent
+        );
+      }
+
+      return () => {
+        setExecuteOutputContent(undefined);
+      };
+    }, [actionDetails, agentId, isFetching]);
+
+    if (isFetching) {
+      return (
+        <EuiSkeletonText size="relative" lines={2} data-test-subj={`${dataTestSubj}-loading`} />
+      );
+    }
+
+    if (!executeOutputContent) {
+      return null;
+    }
```

Comment:
There's a small bug that's caused by these early returns and conditional rendering:
- open Response Console
- perform an execute command
- close the accordions
- open the Response Actions History flyout
- open the newly entered command's details
=> as it opens, the 'Execution output' accordion will open in the Response console, too
![response console](https://user-images.githubusercontent.com/39014407/220978121-eb5e735d-5c67-427d-8ae1-790678be6f56.gif)

This is because the `<ExecutionActionOutputAccordion>` component is conditionally rendered:
- when we open the command details in the flyout, there is a re-fetch of the data
- so `isFetching` will be true
- which means that `<ExecutionActionOutputAccordion>` components will be unmounted and mounted again, therefore starting with the initial opened state

'Pushing down' the loading state can fix this, e.g. something like that (and with removing all early returns):
```typescript
    return (
      <EuiFlexItem data-test-subj={dataTestSubj}>
        <ExecutionActionOutputAccordion
          content={executeOutputContent?.stdout ?? 'loading...'}
          isTruncated={executeOutputContent?.stdoutTruncated ?? false}
          initialIsOpen
          textSize={textSize}
          type="output"
        />
        <EuiSpacer size="m" />
        <ExecutionActionOutputAccordion
          content={executeOutputContent?.stderr ?? 'loading...'}
          isTruncated={executeOutputContent?.stderrTruncated ?? false}
          textSize={textSize}
          initialIsOpen
          type="error"
        />
      </EuiFlexItem>
    );
```
(Still causes a glitch in the '(truncated)' word, though)

Reply:
Yeah, I'm aware of this and will figure out a fix. Thanks for testing this thoroughly 🔥

Reply:
![execute-file](https://user-images.githubusercontent.com/1849116/221646043-bf6f1874-067d-4df7-b2db-63707b549bd9.gif)

Reply:
hey @ashokaditya,
I think the issue is still there - in your video the command you opened in the history was opened in the console. If you open the second one (which is the first in the console, and is closed there), it will open, because of the early return due to `if (isFetching)`

Reply:
Oh right! I missed that again. I'll make a fix.

Reply:
fixed 35911c7f5ac53d1a101d870fbf1fd87cc69eebde

![execute-file-actions-responder](https://user-images.githubusercontent.com/1849116/222129212-7033cb5d-1d00-4720-8b77-43333dc4baa9.gif)